### PR TITLE
Wait for EC2 tags to be available before registering the agent

### PR DESCRIPTION
**Pull request by lox**
This addresses #491. TL;DR is that AWS often takes quite some time to apply tags to a running instance, specifically when in an Autoscaling Group.

**Update**
Based on feedback this has changed to be the default behaviour with a timeout to opt-out of. This means if you enable `tags-from-ec2-tags` then we will wait up to 15 seconds for tags (any tags) to be made available. If you don't set tags, but you set `tags-from-ec2-tags` YOU WILL WAIT 15 SECONDS.

### CODE REVIEW

**Comment by keithpitt** [comment]
Ah, so that was the problem! Here's a question. Is there a scenario where you'd _not_ want that behaviour? It seems that if you've enabled EC2 tags, it's cause you want them. Not because "only get them if you can". Perhaps it should just be the default and not an option?

**Comment by keithpitt** [comment]
Oh I see, you actually don't know if there _are_ any tags. `Wait` actually means "keep asking for tags until you actually get some"

**Comment by lox** [comment]
Yup, it's possible you haven't set any tags for your instance, in which case this would wait forever. I can imagine a future where this is the default if you enabled EC2 tags though.

The other option is to periodically update tags, but I wasn't sure how that would work with re-registering the agent.

File: agent/agent_pool.go [hunk 1]
```diff
@@ -152,9 +157,8 @@ func (r *AgentPool) CreateAgentTemplate() *api.Agent {
 				}
 				s.Break()
 			}
-
 			return err
-		}, &retry.Config{Maximum: 5, Interval: 1 * time.Second, Jitter: true})
+		}, &retry.Config{Maximum: 5, Interval: r.WaitForEC2TagsTimeout / 5, Jitter: true})
```
Comment (keithpitt) [inline agent/agent_pool.go]: Neat!

**Pull request merged**